src/canmatrix/canmatrix.py

```diff
@@ -425,62 +436,75 @@ def pgn(self, value):
 
     @property
     def priority(self):
+        """Get J1939 priority."""
         return self._j1939_prio
 
     @priority.setter
     def priority(self, value):
+        """Set J1939 priority."""
         self._j1939_prio = value
         self.recalcJ1939Id()
 
     @property
     def source(self):
+        """Get J1939 source."""
         return self._j1939_source
 
     @source.setter
-    def pgn(self, value):
+    def source(self, value):
+        """Set J1939 source."""
         self._j1939_source = value
         self.recalcJ1939Id()
 
     def recalcJ1939Id(self):
+        """Recompute J1939 ID"""
         self.id = (self.j1939_source & 0xff) + ((self.j1939_pgn & 0xffff) << 8) + ((self.j1939_prio & 0x7) << 26)
         self.extended = True
         self.is_j1939 = True
 
     #    @property
-#    def cycleTime(self):
-#        if self._cycleTime is None:
-#            self._cycleTime = self.attribute(None, "GenMsgCycleTime")
-#        return self._cycleTime
-
-#    @property
-#    def sendType(self, db = None):
-#        if self._sendType is None:
-#            self._sendType = self.attribute(None, "GenMsgSendType")
- #       return self._sendType
-
-#    @cycleTime.setter
-#    def cycleTime(self, value):
-#        self._cycleTime = value
-#        self.attributes["GenMsgCycleTime"] = value
-
-#    @sendType.setter
-#    def sendType(self, value):
- #       self._sendType = value
- #       self.attributes["GenMsgCycleTime"] = value
+    #    def cycleTime(self):
+    #        if self._cycleTime is None:
+    #            self._cycleTime = self.attribute(None, "GenMsgCycleTime")
+    #        return self._cycleTime
 
-
-
-    def attribute(self, db, attributeName):
-        if attributeName in self.attributes:
-            return self.attributes[attributeName]
+    #    @property
+    #    def sendType(self, db = None):
+    #        if self._sendType is None:
+    #            self._sendType = self.attribute(None, "GenMsgSendType")
+    #       return self._sendType
+
+    #    @cycleTime.setter
+    #    def cycleTime(self, value):
+    #        self._cycleTime = value
+    #        self.attributes["GenMsgCycleTime"] = value
+
+    #    @sendType.setter
+    #    def sendType(self, value):
+    #       self._sendType = value
+    #       self.attributes["GenMsgCycleTime"] = value
+
+    def attribute(self, name, db=None, default=None):
+        """Get any Frame attribute by its name.
+
+        :param str name: attribute name, can be mandatory (ex: id) or optional (customer) attribute.
+        :param CanMatrix db: Optional database parameter to get global default attribute value.
+        :param default: Default value if attribute doesn't exist.
+        :return any: Return the attribute value if found, else `default` or None
+        """
+        if hasattr(self, name):
```

Comment:
Sorry to evolve my request here...  I forgot this was an `attrs` class and I think this will be a touch better.  It will avoid considering methods and such as attributes.

https://www.attrs.org/en/stable/api.html#attr.fields_dict

```python
if name in attr.fields_dict(type(self)):
```

I'll note that this is technically a breaking change given the it will block access via this method to `.attribute` items that shadow the names of regular attributes.  But, I don't think that bothers me.